### PR TITLE
[DNM] Reduce memory copy for IterOptions

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
Signed-off-by: zhangjinpeng1987 <zhangjinpeng@pingcap.com>

## What have you changed? (mandatory)

Reduce one time memory copy for IterOptions' lower_bound and upper_bound.

## What are the type of the changes? (mandatory)

- Improvement (change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

UT

## Does this PR affect documentation (docs) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

NO

## Refer to a related PR or issue link (optional)

- [ ] https://github.com/pingcap/rust-rocksdb/pull/287 is required.

### CODE REVIEW

File: components/engine/src/lib.rs [hunk 1]
```diff
@@ -216,99 +216,70 @@ pub trait Peekable {
...
         upper_bound: Option<Vec<u8>>,
         fill_cache: bool,
     ) -> IterOption {
-        IterOption {
-            lower_bound,
-            upper_bound,
-            prefix_same_as_start: false,
-            fill_cache,
-            seek_mode: SeekMode::TotalOrder,
+        let mut read_options = ReadOptions::new();
```
Comment (breeswish) [inline components/engine/src/lib.rs]: So why not use ReadOptions directly?

File: components/engine/src/lib.rs [hunk 2]
```diff
@@ -216,99 +216,70 @@ pub trait Peekable {
...
-        self.seek_mode == SeekMode::TotalOrder
+    pub fn use_total_order_seek(mut self) -> IterOption {
+        self.0.set_total_order_seek(true);
+        self
     }
 
     #[inline]
-    pub fn lower_bound(&self) -> Option<&[u8]> {
-        self.lower_bound.as_ref().map(|v| v.as_slice())
+    pub fn lower_bound(&self) -> &[u8] {
```
Comment (hicqu) [inline components/engine/src/lib.rs]: What will return if `lower_bound` is not set? Please document it.

File: components/engine/src/lib.rs (hunk 1 above)
Comment (zhangjinpeng1987) [reply components/engine/src/lib.rs]: So we can't use like let opt = IterOptions::new(xxx).xxx().xxx().xxx().

Comment (breeswish) [reply components/engine/src/lib.rs]: Got it. How about calling it `ReadOptionsBuilder`?

**Pull request closed without merging**